File: app/src/main/java/me/anon/model/PlantStage.java
package me.anon.model;

import android.content.Context;
import android.support.annotation.Nullable;

import me.anon.grow.R;

/**
 * // TODO: Add class description
 *
 * @author 7LPdWcaW
 * @documentation // TODO Reference flow doc
 * @project GrowTracker
 */
public enum PlantStage
{
	PLANTED(R.string.planted),
	GERMINATION(R.string.germination),
	SEEDLING(R.string.seedling),
	CUTTING(R.string.cutting),
	VEGETATION(R.string.vegetation),
	FLOWER(R.string.flowering),
	DRYING(R.string.drying),
	CURING(R.string.curing),
	HARVESTED(R.string.harvested);

	private int printString;

	public int getPrintString()
	{
		return printString;
	}

	private PlantStage(int name)
	{
		this.printString = name;
	}

	public static String[] names(Context context)
	{
		String[] names = new String[values().length];
		for (int index = 0; index < names.length; index++)
		{
			names[index] = context.getString(values()[index].getPrintString());
		}

		return names;
	}

	@Nullable
	public static PlantStage valueOfPrintString(Context context, String printString)
	{
		for (PlantStage plantStage : values())
		{
			if (context.getString(plantStage.printString).equals(printString))
			{
				return plantStage;
			}
		}

		return null;
	}
}

File: app/src/main/java/me/anon/model/PlantStage.kt
package me.anon.model

import android.content.Context

import me.anon.grow.R

enum class PlantStage private constructor(val printString: Int)
{
	PLANTED(R.string.planted),
	GERMINATION(R.string.germination),
	SEEDLING(R.string.seedling),
	CUTTING(R.string.cutting),
	VEGETATION(R.string.vegetation),
	FLOWER(R.string.flowering),
	DRYING(R.string.drying),
	CURING(R.string.curing),
	HARVESTED(R.string.harvested);

	companion object
	{
		public fun names(context: Context): Array<String>
		{
			val names = arrayListOf<String>()
			values().forEach { stage ->
				names.add(context.getString(stage.printString))
			}

			return names.toTypedArray()
		}

		public fun valueOfPrintString(context: Context, printString: String): PlantStage?
		{
			for (plantStage in values())
			{
				if (context.getString(plantStage.printString) == printString)
				{
					return plantStage
				}
			}

			return null
		}
	}
}
